src/components/AccountPin.js

```diff
@@ -16,25 +16,57 @@ export default class AccountPin extends Component {
     }
   }
 
+  onNext = () => {
+    const {text} = this.state
+    const {account, extra} = this.props
+
+    this.props.onNextPressed(text, account, extra)
+  }
+
+  onChange = (text) => {
+    this.setState({
+      text
+    })
+  }
+
   render () {
     return (
       <View style={AppStyles.view}>
         <View style={AppStyles.center}>
           <TextInput
-            style={AppStyles.pin}
-            placeholder='enter pin here'
+            autoFocus
+            clearTextOnFocus
             editable
+            fontSize={24}
+            keyboardType='numeric'
             multiline={false}
-            autoFocus
-            returnKeyType='next'
             numberOfLines={1}
-            fontSize={24}
-            onChangeText={(text) => { this.setState({text: text}) }}
+            onChangeText={this.onChange}
+            onEndEditing={this.onNext}
+            placeholder='enter pin here'
+            returnKeyType='next'
+            secureTextEntry
+            style={AppStyles.pin}
             value={this.state.text}
-            onEndEditing={() => { this.props.onNextPressed(this.state.text, this.props.account, this.props.extra) }}
-        />
+          />
+          <View style={[AppStyles.buttonContainer, styles.button]}>
+            <Button
+              onPress={this.onNext}
+              color='green'
+              title='Next'
+              accessibilityLabel='Confrim PIN'
```

Comment:
pressing this button triggers `this.onNext` twice. First time from `onPress` and second time from `onEndEditing`